component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java

```diff
@@ -741,4 +766,28 @@ private HTTPCarbonMessage generateCarbonMessage(List<Header> headers, String con
         httpHeaders.set(HttpConstants.HTTP_METHOD, httpMethod);
         return cMessage;
     }
+
+    private String getPayload(Object payload) {
+        if (HttpConstants.MAP_KEYVALUE.equals(mapType)) {
+            Map<String, Object> params = (HashMap) payload;
+            if (encodePayload) {
+                return params.entrySet().stream()
+                        .map(p -> encodeParameter(p.getKey()) + "=" + encodeParameter(p.getValue()))
+                        .reduce("", (p1, p2) -> p1 + "&" + p2);
+            } else {
+                return params.entrySet().stream()
+                        .map(p -> p.getKey() + "=" + p.getValue())
+                        .reduce("", (p1, p2) -> p1 + "&" + p2);
```

Comment:
I dont think we need to support this case

Reply:
Changed by 92d9c4a7865aedf2fcfbe350231a90b7ed67c4ca